common/unsafe/src/main/java/org/apache/spark/unsafe/memory/HeapMemoryAllocator.java

```diff
@@ -58,7 +58,8 @@ public MemoryBlock allocate(long size) throws OutOfMemoryError {
             final long[] array = arrayReference.get();
             if (array != null) {
               assert (array.length * 8L >= size);
-              MemoryBlock memory = new MemoryBlock(array, Platform.LONG_ARRAY_OFFSET, size);
+              MemoryBlock memory =
+                new OnHeapMemoryBlock(array, Platform.LONG_ARRAY_OFFSET, size);
```

Comment:
`OnHeapMemoryBlock.fromArray`? we should hide `Platform.LONG_ARRAY_OFFSET` from the caller side.

Reply:
sure, done